File: webapp/src/main/java/com/alibaba/intl/livevideo/models/RtpForwardingDestination.java
package com.alibaba.intl.livevideo.models;

/**
 * Destination where Janus can forward RTP data.
 *
 * @author Alibaba Cloud
 */
public class RtpForwardingDestination {

    private String hostname;
    private int audioPort;
    private int videoPort;

    public RtpForwardingDestination() {
    }

    public RtpForwardingDestination(String hostname, int audioPort, int videoPort) {
        this.hostname = hostname;
        this.audioPort = audioPort;
        this.videoPort = videoPort;
    }

    public String getHostname() {
        return hostname;
    }

    public void setHostname(String hostname) {
        this.hostname = hostname;
    }

    public int getAudioPort() {
        return audioPort;
    }

    public void setAudioPort(int audioPort) {
        this.audioPort = audioPort;
    }

    public int getVideoPort() {
        return videoPort;
    }

    public void setVideoPort(int videoPort) {
        this.videoPort = videoPort;
    }

    @Override
    public String toString() {
        return "RtpForwardingDestination{" +
                "hostname='" + hostname + '\'' +
                ", audioPort=" + audioPort +
                ", videoPort=" + videoPort +
                '}';
    }
}

File: webapp/src/main/js/models/RtpForwardingDestination.js
/**
 * Destination where Janus can forward RTP data.
 *
 * @author Alibaba Cloud
 */
export default class RtpForwardingDestination {

    /**
     * @param {{
     *     id: string?,
     *     ipAddress: string?,
     *     audioPort: number?,
     *     videoPort: number?}?
     * } params
     */
    constructor(params) {
        const p = params || {};

        /** @type {string} */
        this.id = p.id || '';

        /** @type {string} */
        this.ipAddress = p.ipAddress || '';

        /** @type {number} */
        this.audioPort = p.audioPort || 0;

        /** @type {number} */
        this.videoPort = p.videoPort || 0;
    }

}